src/libNode/Node.cpp

```diff
@@ -239,6 +250,144 @@ void Node::AddGenesisInfo(SyncType syncType) {
 
+bool Node::ValidateDB() {
+  deque<pair<PubKey, Peer>> dsComm;
+  const string lookupIp = "127.0.0.1";
+  const uint port = 30303;
+
+  for (const auto& dsKey : *m_mediator.m_initialDSCommittee) {
+    dsComm.emplace_back(dsKey, Peer());
+  }
+  std::list<BlockLink> blocklinks;
+  if (!BlockStorage::GetBlockStorage().GetAllBlockLink(blocklinks)) {
+    LOG_GENERAL(WARNING, "BlockStorage skipped or incompleted");
+    return false;
+  }
+
+  blocklinks.sort([](const BlockLink& a, const BlockLink& b) {
+    return std::get<BlockLinkIndex::INDEX>(a) <
+           std::get<BlockLinkIndex::INDEX>(b);
+  });
+
+  std::list<TxBlockSharedPtr> txblocks;
+  if (!BlockStorage::GetBlockStorage().GetAllTxBlocks(txblocks)) {
+    LOG_GENERAL(WARNING, "Fail to get Tx Blocks");
+    return false;
+  }
+
+  txblocks.sort([](const TxBlockSharedPtr& a, const TxBlockSharedPtr& b) {
+    return a->GetHeader().GetBlockNum() < b->GetHeader().GetBlockNum();
+  });
+
+  const auto& latestTxBlockNum = txblocks.back()->GetHeader().GetBlockNum();
+
+  vector<boost::variant<DSBlock, VCBlock, FallbackBlockWShardingStructure>>
+      dirBlocks;
+  for (const auto& blocklink : blocklinks) {
+    if (get<BlockLinkIndex::BLOCKTYPE>(blocklink) == BlockType::DS) {
+      auto blockNum = get<BlockLinkIndex::DSINDEX>(blocklink);
+      if (blockNum == 0) {
+        continue;
+      }
+      DSBlockSharedPtr dsblock;
+      if (!BlockStorage::GetBlockStorage().GetDSBlock(blockNum, dsblock)) {
+        LOG_GENERAL(WARNING, "Could not rertv DS Block " << blockNum);
+        return false;
+      }
+      if (latestTxBlockNum <= dsblock->GetHeader().GetEpochNum()) {
+        break;
+      }
+      dirBlocks.emplace_back(*dsblock);
+
+    } else if (get<BlockLinkIndex::BLOCKTYPE>(blocklink) == BlockType::VC) {
+      auto blockHash = get<BlockLinkIndex::BLOCKHASH>(blocklink);
+      VCBlockSharedPtr vcblock;
+      if (!BlockStorage::GetBlockStorage().GetVCBlock(blockHash, vcblock)) {
+        LOG_GENERAL(WARNING, "Could not retrv VC Block " << blockHash);
+        return false;
+      }
+      if (latestTxBlockNum <= vcblock->GetHeader().GetViewChangeEpochNo()) {
+        break;
+      }
+      dirBlocks.emplace_back(*vcblock);
+    } else if (get<BlockLinkIndex::BLOCKTYPE>(blocklink) == BlockType::FB) {
+      auto blockHash = get<BlockLinkIndex::BLOCKHASH>(blocklink);
+      FallbackBlockSharedPtr fallbackwshardingstruct;
+      if (!BlockStorage::GetBlockStorage().GetFallbackBlock(
+              std::get<BlockLinkIndex::BLOCKHASH>(blocklink),
+              fallbackwshardingstruct)) {
+        LOG_GENERAL(WARNING, "Could not retrv FB blocks " << blockHash);
+        return false;
+      }
+      dirBlocks.emplace_back(*fallbackwshardingstruct);
+    }
+  }
+
+  if (!m_mediator.m_validator->CheckDirBlocks(dirBlocks, dsComm, 0, dsComm)) {
+    LOG_GENERAL(WARNING, "Fail to verify Dir Blocks");
+    return false;
+  }
+
+  vector<TxBlock> txBlocks;
+
+  for (auto txblock : txblocks) {
```

Comment:
use const auto& txBlock

Reply:
the verifier sends a  message  to inform the lookup machine (running on the same machine) that it has done the verification.